polkadot/node/core/approval-voting/src/lib.rs

```diff
@@ -3085,12 +3088,29 @@ where
 			candidate_hash,
 		);
 
+		if let Some(v_idx) = validator_index {
```

Comment:
Mentioned in the other PR as well, but this just tells us we received a vote, but that doesn't mean we use it for approving the candidate, so we need to make sure the approvals we track and reward are only the approvals we use in tranches_to_approve.